### PR TITLE
Create entrypoint iib-add-deprecations [CLOUDDST-23913]

### PR DESCRIPTION
**Pull request by emilyzheng**
The entrypoint calls pubtools-iib-add-deprecations and then signs and published index image.

### CODE REVIEW

**Comment by midnightercz** [comment]
How do you plan to sign produced index images? Shouldn't that be also part of new ep?

**Comment by midnightercz** [comment]
> > How do you plan to sign produced index images? Shouldn't that be also part of new ep?
> 
> It's signed in _sign_and_push (iib_operations.py) which is used by all iib entrypoints.

Right, I overlooked that line in the new method

**Review by midnightercz — APPROVED** [review]

**Review by chandwanitulsi — APPROVED** [review]

File: src/pubtools/_quay/operator_pusher.py [hunk 1]
```diff
@@ -388,6 +388,58 @@ def iib_remove_operators(
             env_vars,
         )
 
+    @classmethod
+    @tx.instrument_func()
+    def iib_add_deprecations(
+        cls,
+        index_image: Optional[str] = None,
```
Comment (chandwanitulsi) [inline src/pubtools/_quay/operator_pusher.py:395]: for add_deprecations index_image should not be optional

Comment (emilyzheng) [reply src/pubtools/_quay/operator_pusher.py]: Thanks. Removed default of it.

**Pull request merged**